### ISSUE TITLE
linter: eslint/prefer-const incorrectly flagging variable in Svelte file

### ISSUE THREAD

**Issue by peterreeves** [oxc-project/oxc#19470]
### What version of Oxlint are you using?

1.48.0

### What command did you run?

 oxlint -c ./.oxlintrc.json

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
    "rules": {
        "prefer-const": ["warn", { "destructuring": "all" }]
    }
}
```
(Truncated to contain only relevant rule)

### What happened?

Given this input Svelte file:
```svelte
<script lang="ts">
    let divEl: HTMLElement | null = $state(null);
</script>

<div bind:this={divEl}></div>
```
Oxlint outputs:
```
  ⚠ eslint(prefer-const): `divEl` is never reassigned.
   ╭─[Example.svelte]
   │<script lang="ts">
   │     let divEl: HTMLElement | null = $state(null);
   ·         ─────
   │ </script>
   ╰────
  help: Use `const` instead.
```
But the variable is actually written to by the `bind:this` part of the template.
Applying Oxlint's suggestion/auto-fix switching from `let` to `const` produces a TypeScript error:
```
Cannot assign to 'divEl' because it is a constant.
```

**Comment by concertypin** [comment]
Maybe this is related: https://github.com/oxc-project/oxc/issues/17991#issuecomment-3773931644
